software/common/libs/framing/rx_buffer.h

```diff
@@ -0,0 +1,36 @@
+/* Copyright 2020-2022, RespiraWorks
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+    http://www.apache.org/licenses/LICENSE-2.0
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+#pragma once
+
+#include <cstddef>
+#include <cstdint>
+
+#include "serial_listeners.h"
+
+/*! \class RxBuffer rx_buffer.h "rx_buffer.h"
```

Comment:
there seem to be a little inconsistency in the use of Doxygen markup. Maybe this is not 
specified in a code standard for this project.

I know there might be an argument for using /* comments in .h files, nevertheless I personally
always use // as all compilers I use support this also for C.

Then /// \class   etc. might be a suggestion